client.go

```diff
@@ -328,8 +375,50 @@ func (a *Client) CopyFromRemotePassThru(ctx context.Context, w io.Writer, remote
 			errCh <- errors.New(res.GetMessage())
 			return
 		}
+		if res.NoStandardProtocolType() {
+			errCh <- errors.New(fmt.Sprintf("Input from server doesn't follow protocol: %s", res.GetMessage()))
+			return
+		}
+
+		fileInfo := NewFileInfos()
+
+		if res.IsTime() {
+			timeInfo, err := res.ParseFileTime()
+
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			fileInfo.Update(timeInfo)
+
+			res, err = ParseResponse(r)
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			if res.IsFailure() {
+				errCh <- errors.New(res.GetMessage())
+				return
+			}
+
+			if res.NoStandardProtocolType() {
+				errCh <- errors.New(fmt.Sprintf("Input from server doesn't follow protocol: %s", res.GetMessage()))
+				return
```

Comment:
This error handling code (line 396 through 408) is getting quite long and duplicates most of the logic. Communicating a failure either through the actual `err` value of `ParseResponse` or through a method `Error()` might help to reduce this. I know you inherited most of that error handling code so to me, this is not blocking for this PR in particular, but it would improve the code quality in my opinion.